### PR TITLE
Add support for app ID named config file, or specifying config file i…

### PR DESCRIPTION
**Pull request by mbaltaks**
…n Info.plist.

### CODE REVIEW

**Comment by mbaltaks** [comment]
Not sure if the Info.plist key should be called `TyphoonDefinitionFilename` or something like `TyphoonConfigFilename`.

**Comment by jasperblues** [comment]
:+1: Looks great. 

The feature that allwos externalizing configuration in an environment specific config file is called 'TyphoonConfig', so the key should be called `TyphoonConfigFilename` rather than `TyphoonDefinitionFilename`.

File: Source/Configuration/Startup/TyphoonStartup.m [hunk 1]
```diff
@@ -65,8 +66,30 @@ + (void)requireInitialFactory
     initialFactoryRequestCount += 1;
 }
 
++ (id<TyphoonDefinitionPostProcessor>)configPostProcessor
+{
+    NSString *fileName = [[NSBundle mainBundle] infoDictionary][@"TyphoonConfigFilename"];
```
Comment (igrekde) [inline Source/Configuration/Startup/TyphoonStartup.m]: Better not to use [NSBundle mainBundle] here - there will be problems with unit tests.

Use - bundleForClass: instead.

**Comment by igrekde** [comment]
Seems like a very nice feature, thanks for contributing!

I have only one suggestion. I think, it's safer to use `-bundleForClass:` instead of `-mainBundle`, otherwise there can be prolems with accessing the config file in unit tests.

**Comment by jasperblues** [comment]
@mbaltaks @igrekde Let's also add a test in the iOS suite. There's an app that hosts the tests - it could be tested that way.

**Comment by mbaltaks** [comment]
Hm, `+ initialFactory` is not being called in TyphoonStartup.m, so the config is not loaded.

**Comment by alexgarbarev** [comment]
@mbaltaks should this config be loaded only when Typhoon loaded from info.plist? Or when from appDelegate too?

**Comment by jasperblues** [comment]
The new feature supports by convention - looks for a file of matching name, as well as explicitly specifying the filen ame in the plist - so technically both. 

And there's an argument towards consistency. 

But: 

- Instantiating initial factory from App Delegate is not a documented feature, and is rarely used. We've talked about removing this. 

- The main use-case is for plist style bootstrapping of iOS apps.

**Comment by igrekde** [comment]
@jasperblues My team is still thinking about using this feature - it seems cool when you have over 30 assemblies of different architecture layers mixed in the Info.plist file.

**Comment by mbaltaks** [comment]
At this point someone with more understanding of the internals will need to help get this working with full autoload. I can only get this working with a call from the assembly class.

**Comment by jasperblues** [comment]
Thanks @mbaltaks 

If anyone wants to grab this, please feel free. Otherwise I'll address as soon as I get through the current heavy amount of client work. 

@igrekde  Introducing you to @mbaltaks, director of Code Monastery - the company that sponsored development of Typhoon Rest Client. . Aleksey and I regularly work with Code Monastery. 

@mbaltaks introducing you to @igrekde, who is a regular Typhoon contributor and evangelist.

**Comment by igrekde** [comment]
Hi, @mbaltaks! Thanks for sponsoring Typhoon Rest Client - it looks great :)

**Comment by igrekde** [comment]
@mbaltaks I've checked the *TyphoonStartup.m*. You should modify the method `+ swizzleSetDelegateMethodOnApplicationClass`:

```
 if (initialFactory)
{
     [self injectInitialFactoryIntoDelegate:delegate];
     [TyphoonComponentFactory setFactoryForResolvingFromXibs:initialFactory];
     [self loadConfig];
}
```

And you can remove this method's call from the `+ initialFactory` method.

@jasperblues I don't see the nice way to test the *TyphoonStartup* logic directly. Do you have any suggestions?

**Comment by jasperblues** [comment]
@igrekde Since the Typhoon test suite is hosted by an (empty) app, we could add the config in that App's plist, then write a test case that checks it via AppDelegate. . . messy, but at least its a test.

**Comment by jasperblues** [comment]
I'm going to merge this in, fix it up as per @igrekde 's suggestion and add a test case.

**Comment by jasperblues** [comment]
Ah the problem was this: 

```objc
id<TyphoonDefinitionPostProcessor> configProcessor = nil;
if ([fileName length]) {
    [TyphoonConfigPostProcessor forResourceNamed:fileName];   <——— this should have been an assignment 
}```

Fixed and created a test case, but had to roll back tests as something went wrong with build. Worked from IDE but not from cmd-line. 

Released as 3.2.6

**Pull request merged**